internal/controller/pipelinerollout/pipelinerollout_progressive.go

```diff
@@ -71,10 +72,19 @@ func (r *PipelineRolloutReconciler) AssessUpgradingChild(
 	assessmentSchedule config.AssessmentSchedule) (apiv1.AssessmentResult, string, error) {
 
 	numaLogger := logger.FromContext(ctx)
-
 	pipelineRollout := rolloutObject.(*apiv1.PipelineRollout)
-
 	childStatus := pipelineRollout.GetUpgradingChildStatus()
+	currentTime := time.Now()
+
+	// Check if endTime has arrived, fail immediately
+	if currentTime.Sub(childStatus.BasicAssessmentStartTime.Time) > assessmentSchedule.End {
+		numaLogger.Debugf("Assessment window ended for upgrading child %s", existingUpgradingChildDef.GetName())
+		_ = progressive.UpdateUpgradingChildStatus(pipelineRollout, func(status *apiv1.UpgradingChildStatus) {
+			status.AssessmentResult = apiv1.AssessmentResultFailure
+			status.BasicAssessmentEndTime = &metav1.Time{Time: currentTime}
+		})
+		return apiv1.AssessmentResultFailure, "Assessment window ended", nil
+	}
```

Comment:
same comment as for monovertex: https://github.com/numaproj/numaplane/pull/844/files#r2289280160